embassy/src/executor/mod.rs

```diff
@@ -56,6 +56,17 @@ impl Spawner {
         }
     }
 
+    /// Used by the `embassy_macros::main!` macro to throw an error when spawn
+    /// fails. This is here to allow conditional use of `defmt::unwrap!`
+    /// without introducing a `defmt` feature in the `embassy_macros` package,
+    /// which would require use of `-Z namespaced-features`.
+    pub fn must_spawn<F>(&self, token: SpawnToken<F>) -> () {
+        #[cfg(feature = "defmt")]
```

Comment:
This should use straight `unwrap!()`, which forwards to defmt or core depending on enabled features. See `src/fmt.rs` .

Reply:
Fixed.